File: HackerRank/Algorithms/1. Warmup/Plus_minus.cpp
#include <iostream>
#include <iomanip>
using namespace std;

int main() {
    /* Enter your code here.
       Read input from STDIN.
       Print output to STDOUT */
    int n;
    cin >> n;
    int arr[n];
    for(int i = 0; i < n; ++i) {
        cin >> arr[i];
    }
    int positive = 0, negative = 0, zero = 0;
    for(int i = 0; i < n; ++i) {
        if (arr[i] > 0) {
            positive++;
        }
        else if (arr[i] < 0) {
            negative++;
        }
        else {
          zero++;
        }
    }
    cout << setprecision(6)  << positive/(float) n << "\n";
    cout << setprecision(6)  << negative/(float) n << "\n";
    cout << setprecision(6)  << zero/(float) n << "\n";
    return 0;
}

File: HackerRank/Algorithms/1. Warmup/Plus_minus.java
import java.util.Scanner;

public class Solution {

    public static void main(String[] args) {
        /* Enter your code here.
           Read input from STDIN.
           Print output to STDOUT.
           Your class should be named Solution. */
        Scanner scan = new Scanner(System.in);
        int n = scan.nextInt();
        int[] arr = new int[n];
        for(int i = 0; i < n; ++i) {
            arr[i] = scan.nextInt();
        }
        scan.close();
        int positive = 0, negative = 0, zero = 0;
        for(int i = 0; i < n; ++i) {
            if (arr[i] > 0) {
                positive++;
            }
            else if (arr[i] < 0) {
                negative++;
            }
            else {
                zero++;
            }
        }
        System.out.printf("%.6f\n", positive/(float)(n));
        System.out.printf("%.6f\n", negative/(float)(n));
        System.out.printf("%.6f\n", zero/(float)(n));
    }
}
